### PR TITLE
feat: Supports uncompression local layer zips in sam local

### PR DESCRIPTION
**Pull request by aahung**
#### Which issue(s) does this change fix?

#947

#### Why is this change necessary?

Support passing a local archive file in "AWS::Serverless::LayerVersion" in local invoke

#### How does it address the issue?

We already handle when function's codeuri points to a zip file, this PR extend that to `ContentUri` of LayerVersion

#### What side effects does this change have?

N/A

#### Checklist

- [x] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [x] Write unit tests
- [ ] Write/update functional tests
- [x] Write/update integration tests
- [x] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: samcli/local/lambdafn/runtime.py [hunk 1]
```diff
@@ -274,13 +276,34 @@ def _get_code_dir(self, code_path):
...
+            decompressed_dir: str = _unzip_file(code_path)
             self._temp_uncompressed_paths_to_be_cleaned += [decompressed_dir]
             return decompressed_dir
 
         LOG.debug("Code %s is not a zip/jar file", code_path)
         return code_path
 
+    def _unarchived_layer(self, layer: Union[str, Dict, LayerVersion]) -> Union[str, Dict, LayerVersion]:
+        """
+        If the layer's content uri points to a supported local archive file, use self._get_code_dir() to
```
Comment (c2tarun) [inline samcli/local/lambdafn/runtime.py:288]: Just wanted to confirm, this only works if layer zip is present on local?

Comment (aahung) [reply samcli/local/lambdafn/runtime.py:288]: yes, same as how we handle `Function` today

Comment (aahung) [reply samcli/local/lambdafn/runtime.py:288]: if customers provide a S3 zip url, it is not gonna work

File: samcli/local/lambdafn/runtime.py [hunk 2]
```diff
@@ -250,7 +252,7 @@ def signal_handler(sig, frame):
         timer.start()
         return timer
 
-    def _get_code_dir(self, code_path):
+    def _get_code_dir(self, code_path: str) -> str:
         """
         Method to get a path to a directory where the Lambda function code is available. This directory will
```
Comment (hawflau) [inline samcli/local/lambdafn/runtime.py:257]: nit: update the docstring to tell that this is also used for Layer

**Review by hawflau — APPROVED** [review]

File: samcli/local/lambdafn/runtime.py (hunk 2 above)
Comment (aahung) [reply samcli/local/lambdafn/runtime.py]: updated 👍

**Review by c2tarun — APPROVED** [review]

**Pull request merged**